src/Core/src/Platform/iOS/SafeAreaPadding.cs

```diff
@@ -0,0 +1,46 @@
+﻿using CoreGraphics;
+using UIKit;
+using System;
+
+namespace Microsoft.Maui.Platform;
+
+internal readonly record struct SafeAreaPadding(double Left, double Right, double Top, double Bottom)
+{
+	public static SafeAreaPadding Empty { get; } = new(0, 0, 0, 0);
+
+	public bool IsEmpty { get; } = Left == 0 && Right == 0 && Top == 0 && Bottom == 0;
+	public double HorizontalThickness { get; } = Left + Right;
+	public double VerticalThickness { get; } = Top + Bottom;
+
+	public CGRect InsetRect(CGRect bounds)
+	{
+		if (IsEmpty)
+		{
+			return bounds;
+		}
+
+		return new CGRect(
```

Comment:
Here, could we include a validation to prevent negative dimensions when padding exceeds bounds? (technically possible, although it probably wouldn't happen).